lib/rack/auth/digest/params.rb

```diff
@@ -17,7 +17,7 @@ def self.dequote(str) # From WEBrick::HTTPUtils
         end
 
         def self.split_header_value(str)
-          str.scan( /(\w+\=(?:"[^\"]+"|[^,]+))/n ).collect{ |v| v[0] }
+          str.scan(/\w+\=(?:"[^\"]+"|[^,]+)/)
```

Comment:
You dropped the [encoding modifier](http://ruby-doc.org/core-2.3.0/Regexp.html#class-Regexp-label-Encoding) also?